### PR TITLE
fix(deps): fix mismatched dependency versions

### PR DESCRIPTION
**Pull request by driev**
**What this PR does / why we need it**:

I was getting some errors and this resolves them.

### CODE REVIEW

File: tests/integration/go.mod [hunk 1]
```diff
@@ -6,8 +6,8 @@ require (
 	github.com/docker/docker v26.1.0+incompatible
 	github.com/onsi/gomega v1.33.1
 	github.com/seldonio/seldon-core/operator/v2 v2.0.0-00010101000000-000000000000
-	k8s.io/api v0.31.2
-	k8s.io/apimachinery v0.31.2
+	k8s.io/api v0.29.2
+	k8s.io/apimachinery v0.29.2
 	k8s.io/client-go v0.29.2
```
Comment (lc525) [inline tests/integration/go.mod:11]: Does upgrading client-go to `v0.31.2` instead lead to failures?

Comment (lc525) [reply tests/integration/go.mod:11]: The versions here should actually be close to the k8s version running in the cluster, or we risk incompatibilities. Therefore perhaps pushing to 31.2 was a bit ambitious.

**Review by lc525 — APPROVED** [review]
lgtm

File: tests/integration/go.mod (hunk 1 above)
Comment (driev) [reply tests/integration/go.mod:11]: And `v0.29.2` matches all of the other modules.

**Pull request merged**